### PR TITLE
ASoC: codec: max98373: Cut the common parts out of soundwire driver.

### PR DESCRIPTION
**Pull request by ryans-lee**
Removed duplicated functions from the soundwire driver.
 - max98373_read_config and max98373_reset were removed.
 - These functions are replaced by max98373_slot_config and max98373_reset in i2c driver.

Removed superfluous volume control from the chip default.

### CODE REVIEW

**Comment by plbossart** [comment]
@ryans-lee can you also fix the SPDX comment for the .h file?

**Comment by RanderWang** [comment]
please check CI failure
````
CHECK: extern prototypes should be avoided in .h files
#189: FILE: sound/soc/codecs/max98373.h:224:
+extern void max98373_reset(struct max98373_priv *max98373, struct device *dev);

CHECK: extern prototypes should be avoided in .h files
#190: FILE: sound/soc/codecs/max98373.h:225:
+extern void max98373_slot_config(struct device *dev,

CHECK: Alignment should match open parenthesis
#191: FILE: sound/soc/codecs/max98373.h:226:
+extern void max98373_slot_config(struct device *dev,
+	struct max98373_priv *max98373);
````

**Comment by ryans-lee** [comment]
> @ryans-lee can you also fix the SPDX comment for the .h file?

Fixed the SPDX comment in max98373.h. Thanks.

**Comment by ryans-lee** [comment]
> please check CI failure
> 
> ```
> CHECK: extern prototypes should be avoided in .h files
> #189: FILE: sound/soc/codecs/max98373.h:224:
> +extern void max98373_reset(struct max98373_priv *max98373, struct device *dev);
> 
> CHECK: extern prototypes should be avoided in .h files
> #190: FILE: sound/soc/codecs/max98373.h:225:
> +extern void max98373_slot_config(struct device *dev,
> 
> CHECK: Alignment should match open parenthesis
> #191: FILE: sound/soc/codecs/max98373.h:226:
> +extern void max98373_slot_config(struct device *dev,
> +	struct max98373_priv *max98373);
> ```

Removed extern in header file. Thanks.

**Comment by ryans-lee** [comment]
> Thanks @ryans-lee ! Aside the checkpatch warnings, looks good. If you can fix those and the SPDX note Pierre raised, this seems good to go.

Thanks for the feedback. Shall fix both checkpatch error and SPDX comment.
[80173a6](https://github.com/thesofproject/linux/commit/80173a64a0668359570f2e78c1484810ccd3f560)

**Comment by plbossart** [comment]
Can we also try to remove the make W=1 warning:

````
/data/pbossart/ktest/sof-dev/sound/soc/codecs/max98373-sdw.c: In function ‘max98373_sdw_dai_hw_params’:
/data/pbossart/ktest/sof-dev/include/linux/bits.h:26:28: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
   26 |   __builtin_constant_p((l) > (h)), (l) > (h), 0)))
      |                            ^
/data/pbossart/ktest/sof-dev/include/linux/build_bug.h:16:62: note: in definition of macro ‘BUILD_BUG_ON_ZERO’
   16 | #define BUILD_BUG_ON_ZERO(e) ((int)(sizeof(struct { int:(-!!(e)); })))
      |                                                              ^
/data/pbossart/ktest/sof-dev/include/linux/bits.h:39:3: note: in expansion of macro ‘GENMASK_INPUT_CHECK’
   39 |  (GENMASK_INPUT_CHECK(h, l) + __GENMASK(h, l))
      |   ^~~~~~~~~~~~~~~~~~~
/data/pbossart/ktest/sof-dev/sound/soc/codecs/max98373-sdw.c:627:25: note: in expansion of macro ‘GENMASK’
  627 |   port_config.ch_mask = GENMASK(stream_config.ch_count - 1, 0);
      |                         ^~~~~~~
/data/pbossart/ktest/sof-dev/include/linux/bits.h:26:40: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
   26 |   __builtin_constant_p((l) > (h)), (l) > (h), 0)))
      |                                        ^
/data/pbossart/ktest/sof-dev/include/linux/build_bug.h:16:62: note: in definition of macro ‘BUILD_BUG_ON_ZERO’
   16 | #define BUILD_BUG_ON_ZERO(e) ((int)(sizeof(struct { int:(-!!(e)); })))
      |                                                              ^
/data/pbossart/ktest/sof-dev/include/linux/bits.h:39:3: note: in expansion of macro ‘GENMASK_INPUT_CHECK’
   39 |  (GENMASK_INPUT_CHECK(h, l) + __GENMASK(h, l))
      |   ^~~~~~~~~~~~~~~~~~~
/data/pbossart/ktest/sof-dev/sound/soc/codecs/max98373-sdw.c:627:25: note: in expansion of macro ‘GENMASK’
  627 |   port_config.ch_mask = GENMASK(stream_config.ch_count - 1, 0);
      |                         ^~~~~~~
````

Thanks!

File: sound/soc/codecs/max98373.h [hunk 1]
```diff
@@ -221,4 +221,7 @@ struct max98373_priv {
 };
 
 extern const struct snd_soc_component_driver soc_codec_dev_max98373_sdw;
+extern void max98373_reset(struct max98373_priv *max98373, struct device *dev);
+extern void max98373_slot_config(struct device *dev,
+	struct max98373_priv *max98373);
```
Comment (kv2019i) [inline sound/soc/codecs/max98373.h]: Please drop the "extern" from this patch.

File: sound/soc/codecs/max98373.h [hunk 2]
```diff
@@ -220,8 +221,8 @@ struct max98373_priv {
 	unsigned int rx_mask;
 };
 
-extern const struct snd_soc_component_driver soc_codec_dev_max98373_sdw;
-extern void max98373_reset(struct max98373_priv *max98373, struct device *dev);
-extern void max98373_slot_config(struct device *dev,
-	struct max98373_priv *max98373);
+const struct snd_soc_component_driver soc_codec_dev_max98373_sdw;
```
Comment (kv2019i) [inline sound/soc/codecs/max98373.h]: .. and keep "extern" for soc_codec_dev_max98373_sdw. It is still needed.

Comment (ryans-lee) [reply sound/soc/codecs/max98373.h]: > Can we also try to remove the make W=1 warning:
> 
> ```
> /data/pbossart/ktest/sof-dev/sound/soc/codecs/max98373-sdw.c: In function ‘max98373_sdw_dai_hw_params’:
> /data/pbossart/ktest/sof-dev/include/linux/bits.h:26:28: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
>    26 |   __builtin_constant_p((l) > (h)), (l) > (h), 0)))
>       |                            ^
> /data/pbossart/ktest/sof-dev/include/linux/build_bug.h:16:62: note: in definition of macro ‘BUILD_BUG_ON_ZERO’
>    16 | #define BUILD_BUG_ON_ZERO(e) ((int)(sizeof(struct { int:(-!!(e)); })))
>       |                                                              ^
> /data/pbossart/ktest/sof-dev/include/linux/bits.h:39:3: note: in expansion of macro ‘GENMASK_INPUT_CHECK’
>    39 |  (GENMASK_INPUT_CHECK(h, l) + __GENMASK(h, l))
>       |   ^~~~~~~~~~~~~~~~~~~
> /data/pbossart/ktest/sof-dev/sound/soc/codecs/max98373-sdw.c:627:25: note: in expansion of macro ‘GENMASK’
>   627 |   port_config.ch_mask = GENMASK(stream_config.ch_count - 1, 0);
>       |                         ^~~~~~~
> /data/pbossart/ktest/sof-dev/include/linux/bits.h:26:40: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
>    26 |   __builtin_constant_p((l) > (h)), (l) > (h), 0)))
>       |                                        ^
> /data/pbossart/ktest/sof-dev/include/linux/build_bug.h:16:62: note: in definition of macro ‘BUILD_BUG_ON_ZERO’
>    16 | #define BUILD_BUG_ON_ZERO(e) ((int)(sizeof(struct { int:(-!!(e)); })))
>       |                                                              ^
> /data/pbossart/ktest/sof-dev/include/linux/bits.h:39:3: note: in expansion of macro ‘GENMASK_INPUT_CHECK’
>    39 |  (GENMASK_INPUT_CHECK(h, l) + __GENMASK(h, l))
>       |   ^~~~~~~~~~~~~~~~~~~
> /data/pbossart/ktest/sof-dev/sound/soc/codecs/max98373-sdw.c:627:25: note: in expansion of macro ‘GENMASK’
>   627 |   port_config.ch_mask = GENMASK(stream_config.ch_count - 1, 0);
>       |                         ^~~~~~~
> ```
> 
> Thanks!

Shall fix W=1 warning. Thanks!

File: sound/soc/codecs/max98373.h (hunk 1 above)
Comment (ryans-lee) [reply sound/soc/codecs/max98373.h]: Thanks! Shall fix this.

**Comment by RanderWang** [comment]
please check CI. Thanks!
````
Commit 0153a525439c ("ASoC: codec: max98373: Cut the common parts out of soundwire driver")
-------------------------------------------------------------------------------------------
CHECK: Alignment should match open parenthesis
#167: FILE: sound/soc/codecs/max98373.c:904:
+void max98373_slot_config(struct device *dev,
 	struct max98373_priv *max98373)
````

File: sound/soc/codecs/max98373.h [hunk 3]
```diff
@@ -1,5 +1,6 @@
-// SPDX-License-Identifier: GPL-2.0
-// Copyright (c) 2017, Maxim Integrated
+/* SPDX-License-Identifier: GPL-2.0-only
```
Comment (plbossart) [inline sound/soc/codecs/max98373.h]: the ask is that the first line is in one comment, then a second comment for copyright, etc.

**Comment by ryans-lee** [comment]
> please check CI. Thanks!
> 
> ```
> Commit 0153a525439c ("ASoC: codec: max98373: Cut the common parts out of soundwire driver")
> -------------------------------------------------------------------------------------------
> CHECK: Alignment should match open parenthesis
> #167: FILE: sound/soc/codecs/max98373.c:904:
> +void max98373_slot_config(struct device *dev,
>  	struct max98373_priv *max98373)
> ```

Shall fix this. Thanks.

File: sound/soc/codecs/max98373.h (hunk 3 above)
Comment (ryans-lee) [reply sound/soc/codecs/max98373.h]: Let me correct it as followed. Thanks.
/* SPDX-License-Identifier: GPL-2.0-only */
/* Copyright (c) 2017 Maxim Integrated */

**Comment by ryans-lee** [comment]
Thanks for the review!

**Pull request merged**